src/main/java/org/spongepowered/common/event/SpongeCommonEventFactory.java

```diff
@@ -1786,4 +1788,43 @@ public static ItemStack throwDropItemAndConstructEvent(final net.minecraft.entit
         return event;
     }
 
+    /**
+     * Common method to fire a LaunchProjectileEvent and populate its context, then calls SpawnEntityEvent.
+     *
+     * This must be called in the main thread.
+     *  @param frame a frame to populate the context and use for the event.
+     * @param player the player who launched the projectiles.
+     * @param projectiles the launched projectiles.
+     * @param context the context in which this method is called.
+     */
```

Comment:
Gonna need a `@SuppressWarnings({"rawTypes", "uncheckedCast"})` or something for the raw `List` casting going on.